### DISCUSSION TITLE
[Support]: Main stream is lost / not recorded during events, but preview stream is still recorded

### DISCUSSION THREAD

**Discussion by DanGamingTV** [blakeblackshear/frigate#15573] [General Support]

**Comment by NickM-27** [comment]
Are the recordings you're viewing after the 7 days? All recordings should be kept for 7 days so if they're not then that's likely going to be a camera issue.

**Comment by hawkeye217** [comment]
Along with what @NickM-27 said, you'll want to double check your `input_args`. You're using a preset designed for use with go2rtc. 

We'll need to see logs to be able to help further as well.

**Comment by DanGamingTV** [comment]
The footage I was viewing in the first example was within the past 7 days.
I changed my input_args to use the generic rtsp preset.
[New config](https://gist.github.com/DanGamingTV/3ab3a6e1df421ce9237aa88f0cce8fe4)
I observed the issue happening again: [Video](https://youtu.be/wyT8LVmG9gA)
There were no logs created during the event.
I am noticing that this skipping only happens when there are at least two objects in frame (e.g a person and a car).

Netdata screenshots during the event:
![Screenshot 2024-12-23 at 21-21-22 Tower netdata dashboard](https://github.com/user-attachments/assets/5c434273-2662-45c6-abc5-cb059df6a8fd)
![Screenshot 2024-12-23 at 21-21-30 Tower netdata dashboard](https://github.com/user-attachments/assets/785bc8cc-8c48-4c24-a5f4-348fc316a80a)
![Screenshot 2024-12-23 at 21-21-35 Tower netdata dashboard](https://github.com/user-attachments/assets/801e6646-ad46-48e2-8c8b-e3a711e3ec3d)
![Screenshot 2024-12-23 at 21-21-41 Tower netdata dashboard](https://github.com/user-attachments/assets/4c8b1ce5-756c-4768-af92-44f0bec3ae0a)
![Screenshot 2024-12-23 at 21-21-48 Tower netdata dashboard](https://github.com/user-attachments/assets/6d89de6c-3da5-4335-b6f8-3baf09be8204)

**Reply by NickM-27** [reply]
Not able to watch the video. Also, I'd suggest checking the recordings folders directly for the time to see if any time is missing (remember folders are in UTC)

**Reply by DanGamingTV** [reply]
Apologies, the [video link](https://youtu.be/wyT8LVmG9gA) should work now. Here's the files in the recordings folder for that camera during that event:
![image](https://github.com/user-attachments/assets/5a3116cf-e2a4-431d-aae7-2977f7b11dbb)
There are 346 total video files in this folder, which is less than the 360 I'd expect it to be around (360 x 10 second segments per hour).
![image](https://github.com/user-attachments/assets/94048325-76e3-4fba-98ac-cb30a3773c10)

Here's what happens when I attempt to export this event:
![image](https://github.com/user-attachments/assets/65b25053-d47f-4721-8e3a-5e33df2c6c71)
![image](https://github.com/user-attachments/assets/9a5a9c27-a769-47e7-8596-234fa02815db)
Only 1 second of video gets exported.
And oddly, there is no error log saying that footage was missing when doing the export:
![image](https://github.com/user-attachments/assets/ffa7a0b8-f13c-4356-94d2-aa310c1de551)

**Reply by NickM-27** [reply]
Okay, so we can see in the files there is a large gap where you'd expect to see this occurring, so the recordings were not saved. 

At this point I think I would suggest updating to 0.15-beta4 as we have reworked some of the recording retention logic and I'd rather see if this still occurs in that version.

**Comment by DanGamingTV** [comment]
Hey guys, just a small update. I've switched over to a USB Coral for detection. This issue has sprung up again, in a situation where there are two objects being detected at the same time / within a very close window of each other.
My config is set up to record 24/7, and to retain 7 days worth of footage. However, I also had a retention of 31 days set up for events:
```yaml
record:
  enabled: true
  retain:
    days: 7
    mode: all
  events:
    pre_capture: 25
    post_capture: 20
    retain:
      default: 31
      mode: all
```
Is it possible that having 24/7 recording and event recording set up at the same time is causing errors with the recording maintainer? e.g. when an event occurs the camera feed gets recorded twice - for the 24/7 recording and for the event recording, leading to issues.
Just a thought :)

**Reply by NickM-27** [reply]
Nothing is ever recorded twice, if you have continuous recording enabled then it doesn't consider the event retention until cleanup time
